substrate/frame/nis/src/lib.rs

```diff
@@ -190,18 +194,23 @@ pub mod pallet {
 		},
 		PalletId,
 	};
-	use frame_system::pallet_prelude::*;
+	use frame_system::pallet_prelude::{BlockNumberFor as SystemBlockNumberFor, *};
 	use sp_arithmetic::{PerThing, Perquintill};
 	use sp_runtime::{
-		traits::{AccountIdConversion, Bounded, Convert, ConvertBack, Saturating, Zero},
+		traits::{
+			AccountIdConversion, BlockNumberProvider, Bounded, Convert, ConvertBack, Saturating,
+			Zero,
+		},
 		Rounding, TokenError,
 	};
 
-	type BalanceOf<T> =
+	pub type BlockNumberFor<T> =
+		<<T as Config>::BlockNumberProvider as BlockNumberProvider>::BlockNumber;
+	pub type BalanceOf<T> =
```

Comment:
I don't mind reexporting types, but those names are so common I am afraid it will crate unecessary conflicts and bad IDE suggestions.
```suggestion
	pub(crate) type BlockNumberFor<T> =
		<<T as Config>::BlockNumberProvider as BlockNumberProvider>::BlockNumber;
	pub(crate) type BalanceOf<T> =
```